cpp/include/cugraph/detail/shuffle_wrappers.hpp

```diff
@@ -200,6 +200,33 @@ shuffle_int_vertex_value_pairs_to_local_gpu_by_vertex_partitioning(
   rmm::device_uvector<value_t>&& values,
   raft::host_span<vertex_t const> vertex_partition_range_lasts);
 
+/**
+ * @ingroup shuffle_wrappers_cpp
+ * @brief Shuffle vertices using the internal vertex key function which returns the target GPU ID.
+ *
+ * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
+ * @tparam value1_t Type of first vertex values. Needs to be an integral type.
+ * @tparam value2_t Type of second vertex values. Needs to be an integral type.
+ *
+ * @param[in] handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator,
+ * @param[in] vertices Vertex IDs to shuffle
+ * @param[in] values1 First Vertex Values to shuffle
+ * @param[in] values2 Second Vertex Values to shuffle
+ * @param[in] vertex_partition_range_lasts From graph view, vector of last vertex id for each gpu
+ *
+ * @return tuple containing device vector of shuffled vertices and device vectors of corresponding
+ *         values
+ */
+template <typename vertex_t, typename value1_t, typename value2_t>
+std::
+  tuple<rmm::device_uvector<vertex_t>, rmm::device_uvector<value1_t>, rmm::device_uvector<value2_t>>
+  shuffle_int_vertex_two_value_pairs_to_local_gpu_by_vertex_partitioning(
+    raft::handle_t const& handle,
+    rmm::device_uvector<vertex_t>&& vertices,
+    rmm::device_uvector<value1_t>&& values1,
+    rmm::device_uvector<value2_t>&& values2,
+    raft::host_span<vertex_t const> vertex_partition_range_lasts);
```

Comment:
Should we really create this function? I think this can be handled by explicitly instantiating `shuffle_int_vertex_value_pairs_to_local_gpu_by_vertex_partitioning` with `value_t=thrust::tuple<value1_t, value2_t>`.